src/commands/importDocuments.ts

```diff
@@ -0,0 +1,134 @@
+import { ext } from '../extensionVariables';
+import { MongoCollectionTreeItem } from '../mongo/tree/MongoCollectionTreeItem';
+
+export async function importDocuments(tree: AzureTreeDataProvider, uris: vscode.Uri[] | undefined, collectionNode: IAzureParentNode<MongoCollectionTreeItem | DocDBCollectionTreeItem> | undefined): Promise<void> {
+    if (!uris) {
+        uris = await askForDocuments();
+    }
+    let ignoredUris: vscode.Uri[] = []; //account for https://github.com/Microsoft/vscode/issues/59782
+    uris = uris.filter((uri) => {
+        if (uri.fsPath.endsWith('.json')) {
+            return true;
+        } else {
+            ignoredUris.push(uri);
+            return false;
+        }
+    });
+    if (ignoredUris.length) {
+        ext.outputChannel.appendLine(`The following selected files are not json: ${ignoredUris.map(uri => uri.fsPath).join(',')}. \nIgnoring these.`);
+        ext.outputChannel.show();
+    }
+    if (!collectionNode) {
+        collectionNode = <IAzureParentNode<MongoCollectionTreeItem | DocDBCollectionTreeItem>>await tree.showNodePicker([MongoCollectionTreeItem.contextValue, DocDBCollectionTreeItem.contextValue]);
+    }
+
+    const documents = await vscode.window.withProgress(
+        {
+            location: vscode.ProgressLocation.Notification,
+            title: "Import documents..."
+        },
+        (_progress) => parseDocumentsForErrors(uris)
+    );
+
+    let result: string;
+    if (collectionNode.treeItem instanceof MongoCollectionTreeItem) {
+        const collectionTreeItem = <MongoCollectionTreeItem>collectionNode.treeItem;
+        result = await collectionTreeItem.executeCommand('insertMany', [JSON.stringify(documents)]);
+    } else {
+        result = await insertDocumentsIntoDocdb(<IAzureParentNode<DocDBCollectionTreeItem>>collectionNode, documents, uris);
+    }
+    await collectionNode.refresh();
+    await vscode.window.showInformationMessage(result);
+}
+
+async function askForDocuments(): Promise<vscode.Uri[]> {
+    return await ext.ui.showOpenDialog({
+        canSelectMany: true,
+        openLabel: "Import",
+        filters: {
+            "JSON": ["json"]
+        },
+        defaultUri: vscode.Uri.file(vscode.workspace.rootPath)
+    });
+}
+
+// tslint:disable-next-line:no-any
+async function parseDocumentsForErrors(nodes: vscode.Uri[]): Promise<any[]> {
+    const parseResult = await parseDocuments(nodes);
+    const documents = parseResult[0];
+    const hasErrors: boolean = parseResult[1];
+    if (hasErrors) {
+        throw new Error(`Errors found in some documents.\nPlease see the output, fix these and try again.`);
+    }
+    return documents;
+}
+
+// tslint:disable-next-line:no-any
+async function parseDocuments(uris: vscode.Uri[]): Promise<[any[], boolean]> {
+    let documents = [];
+    let errors = {};
+    let errorFoundFlag: boolean = false;
+    for (let uri of uris) {
+        const text: string = await fse.readFile(uri.fsPath, 'utf-8');
```

Comment:
Actually I think we want to use `fse.readJson` instead of `fse.readFile` and `JSON.parse`

For example, change the encoding on a file from 'utf-8' to something else (I did 'utf-8 with bom') from the status bar in VS Code:
![screen shot 2018-10-04 at 10 25 42 am 2](https://user-images.githubusercontent.com/11282622/46491577-f3eee200-c7bf-11e8-9d59-ed6016113d0e.png)

fse.readJson will automatically detect the encoding and parse appropriately (based on my testing), but the current code will fail with an error like this:
![screen shot 2018-10-04 at 10 27 37 am](https://user-images.githubusercontent.com/11282622/46491643-213b9000-c7c0-11e8-99db-3f80cff15e62.png)